<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <node pkg="snap_cam" type="snap_cam_exec" name="front_camera" output="screen">

    <!-- resolution: QVGA, VGA, 720p, 1080p, 4k -->
    <param name="camera_resolution" value="4k"/>

    <!-- auto, infinity, macro, continuous-video, continuous-picture, manual -->
    <param name="focus_mode" value="continuous-video"/>

    <!-- auto, incandescent, fluorescent, warm-fluorescent, daylight, cloudy-daylight, twilight, shade, manual-cct -->
    <param name="white_balance" value="auto"/>

    <!-- auto, ISO_HJR, ISO100, ISO200, ISO400, ISO800, ISO1600, ISO3200 -->
    <param name="iso" value="ISO3200"/>

    <!-- yuv420sp, yuv420p, nv12-venus, bayer-rggb -->
    <param name="preview_format" value="yuv420sp"/>

    <!-- brightness: integer beween 1-6-->
    <param name="brightness" value="3"/>

    <!-- sharpness: integer between 0-36 in increments of 6-->
    <param name="sharpness" value="18"/>

    <!-- contrast: integer between 0-10 -->
    <param name="contrast" value="5"/>

    <!-- exposure: integer between 0 and 511 --> 
    <param name="exposure" value="100"/>

    <!-- gain: integer between 0 and 255 -->
    <param name="gain" value="50"/>

    <!-- attitude_topic: Name of the topic publishing attitude data for this quad -->
    <param name="attitude_topic" value="/phoenix/Attitude2D"/>

    <!-- position_topic: Name of the topic publishing position data for this quad -->
    <param name="position_topic" value="/phoenix/SingleBaselineRTK"/>

    <!-- balloon_position_topic: Name of topic publishing the position of the balloon -->
    <param name="balloon_topic" value="/phoenix/balloon"/>

    <!-- Sensor Params. Defined in sensorParams.h -->
    <rosparam param="rcB">[0.0, 0.0146, -0.0907]</rosparam>
    <rosparam param="rpB">[0.0, 0.0, 0.0]</rosparam>
    <rosparam param="rrG">[-741990.536, -5462227.638, 3198019.45]</rosparam>
    <rosparam param="riG">[-742015.136, -5462219.538, 3198014.35]</rosparam>
    <param name="f" value="1656.06"/>
    <param name="k1" value="-0.0269765"/>

    </node>
</launch>
